### ISSUE TITLE
Tooltip Persists on iOS (Safari Mobile)

### ISSUE THREAD

**Issue by braxtonc** [recharts/recharts#1109]
## Do you want to request a feature or report a bug?
BUG
## What is the current behavior?
Opening Tooltip works as expected but the Tooltip persists as you open other tooltips
the reason being that iOS devices do not have onMouseLeave.
Recharts Tooltip persists on iOS in recharts src code you can see it uses onMouseLeave

[https://github.com/recharts/recharts/blob/master/src/chart/generateCategoricalChart.js#L713](url)

this presents a problem on (iPhone/iPad) because Safari doesn't support mouseleave on iOS more on that subject here

[https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html](url)

Are there any tricks I can do to make the tooltip behave the way it does on other browsers? Below I have attached a screenshot of the issue.
![screen shot 2017-12-20 at 11 22 32 am_censored](https://user-images.githubusercontent.com/28901155/34223194-a766c87a-e57b-11e7-919d-c1f1ed195a76.jpg)

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar (template: http://jsfiddle.net/ndLhnegs/).
you must test on a device that is running iOS or you can use a Xcode simulator testing in chrome will not work because it has access to the needed onMouseLeave
[http://jsfiddle.net/kuddde3q/](url)

## What is the expected behavior?
after opening a tooltip on a chart when you open a tooltip on a different chart the original closes

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?
"recharts": "^1.0.0-beta.1"
iOS, Safari Mobile(seems to be all of them that I have tested)
not sure if this was an issue in older versions of recharts.

**Comment by braxtonchristensen** [comment]
@xile611 Would you like me to submit a PR for this?

**Comment by chris-guidry** [comment]
I would like to resolve this issue as well. Any touch event outside of the chart should clear the tooltip. In Android this works fine (maybe it translates a new touch into a mouseleave?), but in iOS it can only be cleared if there is a touch event within the chart.

It seems like there would need to be a document.addEventListener to detect a touchstart and trigger mouseleave.

@braxtonc - did you have an approach in mind for fixing the problem?

**Comment by cainianhua** [comment]
@xile611 How do you think this issue? Can you help to explain what do you think? Thanks in advance.

**Comment by willm78** [comment]
One option if you have multiple charts that share the same x-axis is to pass an identical `syncId` prop to each of the charts. That should prevent a visible `Tooltip` from persisting on multiple charts simultaneously.

I came across this issue while looking for a way to automatically hide the `Tooltip` on every `onTouchEnd` event. If anyone's interested in a potential solution to this, I ended up using a ref. It seems to work fine. Unfortunately my solution also hides the `Tooltip` on an actual `onMouseUp` event too, but I'm ok with that for now. I did something like this.

```js
<LineChart
  ref={this.chartRef}
  data={data}
  onMouseUp={(mouse, e) => {
    this.chartRef.current.handleMouseLeave(e);
  }}
>
  {...children}
</LineChart>
```
